docs/guides/python/llama-rag.mdx

```diff
@@ -0,0 +1,293 @@
+---
+description: 'Making LLMs smarter with Dynamic Knowledge Access using Retrieval Augmented Generation'
+tags:
+  - API
+  - AI & Machine Learning
+languages:
+  - python
+featured:
+  image: /docs/images/guides/llama-rag/featured.png
+  image_alt: 'Llama RAG featured image'
+published_at: 2024-11-15
+updated_at: 2024-11-15
+---
+
+# Making LLMs smarter with Dynamic Knowledge Access
+
+This guide shows how to use Retrieval Augmented Generation (RAG) to enhance a large language model (LLM). RAG is the process of enabling an LLM to reference context outside of its initial training data before generating its response. It can be extremely expensive in both time and computing power to train a model that is useful for your own domain-specific purposes. Therefore, using RAG is a cost-effective alternative to extending the capabilities of an existing LLM.
+
```

Comment:
```suggestion
To demonstrate RAG in this guide, we'll provide Llama 3.2 with access to Nitric's documentation so that it can answer specific questions. You can adjust this guide with another data source that meets your needs.

```

Reply:
Feel free to edit this - I just think it's useful to give an idea of specifically what's in the guide.